core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -474,13 +474,18 @@ class TimeSeriesShard(val dataset: Dataset,
     val result: Set[ZeroCopyUTF8String] = new mutable.HashSet[ZeroCopyUTF8String]()
     val partIterator = partKeyIndex.partIdsFromFilters(filter, startTime, endTime)
     while(partIterator.hasNext && result.size < limit) {
-      val nextPart = getPartitionFromPartId(partIterator.next())
-      val consumer = new SeqIndexValueConsumer(indexName)
-      // FIXME This is non-performant and temporary fix for fetching label values based on filter criteria.
-      // Other strategies needs to be evaluated for making this performant - create facets for predefined fields or
-      // have a centralized service/store for serving metadata
-      dataset.partKeySchema.consumeMapItems(nextPart.partKeyBase, nextPart.partKeyOffset, 0, consumer)
-      result ++= consumer.labelValues
+      val partId = partIterator.next()
+      val nextPart = getPartitionFromPartId(partId)
+      if (nextPart != UnsafeUtils.ZeroPointer) {
+        val consumer = new SeqIndexValueConsumer(indexName)
+        // FIXME This is non-performant and temporary fix for fetching label values based on filter criteria.
+        // Other strategies needs to be evaluated for making this performant - create facets for predefined fields or
+        // have a centralized service/store for serving metadata
+        dataset.partKeySchema.consumeMapItems(nextPart.partKeyBase, nextPart.partKeyOffset, 0, consumer)
```

Comment:
This needs to be almost immediately fixed or else non-Prometheus schemas will break, in a very ugly and non obvious way.  (ie possibly SEGV's or hangs).

Reply:
Agreed.